### ISSUE TITLE
Remove examples directory and move data directory to notebooks

### ISSUE THREAD

**Issue by DinoBektesevic** [dirac-institute/kbmod#328]
I've been looking at the `examples` directory and I don't understand it. Last significant touch-up, that isn't black auto-formatting, is `b2bb6c4` from 2018 and it doesn't run out of the box. 

I vote we remove it. 

I propose we move `data` to `notebooks`. This paves the way to separating data, examples, and notebooks from code search functionality into a separate repo. This is good because the code in the repo is: 

```bash
468K	./src
620K	./tests
12K	./.github
368K	./docs
```

and keeping these files around 

```bash
8.1M	./data
6.3M	./notebooks
```

which balloons our git history every-time the data is edited/changed:

```bash
121M	./.git
```

So at the moment we're pulling in nearly 100MB at every clone for no good reason. First 20 files are all stored changes on either a jupyter notebook or a data file (sizes in kb, `pack` means size once compressed) :

```
size   pack  SHA                                       location
15436  6072  41e3e9aec6efaa9353d567183b1d1a51f40c9444  data/postage_stamp_training.dat
7897   2427  2bea572b45af52ac46c7fd025a20bd1a7d861bb0  code/kbmodGPU/pybinds/psf.so
5827   1736  6d79f323b5fcfba8050411979c14e2c1bf47eeb6  lib/libcfitsio.a
5636   1643  a37c591c6b1a4bdc37e6545615beaf8be1a76a3c  search/pybinds/kbmod.cpython-35m-x86_64-linux-gnu.so
5601   1881  f2013b75edf861a3f2f96ca666c2302aacceb5ed  search/pybinds/kbmod.cpython-35m-x86_64-linux-gnu.so
4737   1532  35bffd7e0c2eb8219dac652ccabdcb5cf9fa0034  code/kbmodGPU/Debug/kbmodGPU
3877   1445  cf34c1683564cca938e21a586e42a466355f6e0c  code/gpu/debug/CudaTracker
3433   2562  661a54836c6031165a1f5dfdc84b8ef28d76f784  notebooks/orbit_fitting_demo.ipynb
3085   3086  225fa5c51517fc3060764defde4484fc9410153a  data/mpc_hits_asteroids.csv.gz
2989   2175  07c46188fa4e86eb3ef2c8e0008ecaedf4bca20a  notebooks/kbmod_neo_analysis_demo.ipynb
2917   1102  d108ed16538852e2b07829c84c276e0e240be52f  analysis/classification/asteroids.ipynb
2543   1797  5cac7aae247f3a7b3fa13635b63189fe0964c377  notebooks/kbmod_neo_analysis_demo.ipynb
2523   1893  a24b828f2d458a642e6f84188e79ad9642d4af5b  notebooks/.ipynb_checkpoints/kbmod_CNN-checkpoint.ipynb
2522   1892  721d810b72a22d14058c1b451253704bc1a3058a  notebooks/kbmod_CNN.ipynb
2250   1586  cad4cc3d725377d6f22168a6391d4bba1630ced3  notebooks/HITS_Main_Belt_Example.ipynb
2185   453   3e6407350b4eca33eef17d1b7184955ba1770b50  code/gpu/debug/results.txt
2185   545   7a01d04df79561ac9393c16047197a7d86f7a979  code/gpu/debug/results.txt
2180   480   eb4f5c3b2e4218bff99abad4a273289957e2a375  search/Debug/results.txt
2082   637   8407c1fe0a72a23d11ba3fe8a0174c7a9fe14bff  code/kbmodGPU/tests/results.txt
2082   638   85ea4299e5e924cad498a48007662cb5737b31fc  tests/test.txt
```

Eventually I want to prune these from history and tracking and cut down on the size of the repo.

**Comment by cchris28** [comment]
Are we cleaning the output from notebooks before we commit them? I have noticed that running the notebooks to create the sphinx documentation without cleaning them afterwards causes a lot of bloat and since it changes with each run it would quickly add up.

**Comment by jeremykubica** [comment]
I don't think we are clearing the notebooks before committing them. Or at least I have not been.

The examples code is left-over from earlier versions of KBMOD. I will send a PR to remove it.

**Comment by DinoBektesevic** [comment]
Ah, I see, well we can have the `data` dir where it is, I just assumed it was all for the notebooks.

The `data/demo` is the largest part of the directory so if it's not worth moving it, it's probably not worth moving out the notebooks yet. I don't have any GPU machines available to me right now to test the time it takes to generate the images, but from what I've seen I'm pretty sure its not that bad. 

In principle I don't mind keeping the `data` dir, these files rarely change so they don't pollute the git history but we should check in on the size once in a while. If we want to force resolution of the issue, long term it would be good to move the notebooks out I guess because for them this will keep happening every-time we update the interface/functionality, we can resolve these issues in few ways: 

* no split, we trim the notebooks, run them before doc build to populate output fields, will help save a big majority of space as it's the plots that take it up would be my guess
* we split, tests generate the fits they need when they need it, notebooks package the pre-generated ones. 
* we punt this to later -  majority of issue, seems to me, is the interface demands a file and is difficult to use otherwise, since we're working on this as we speak we could make sure the new interface is not coupled so strongly to the FITS file and takes in other things. Then our tests don't need pre-generated files and it's down to optimizing the time it takes to create the fake data. If the image generation is not fast enough we can take a look at that. Looking at it it shouldn't be that slow, but I'm sure we can trim down on some for loops.
 
This issue is not a priority, so I'm in no rush to deal with it at the moment, just wanted a record to remember later.
